src/Trakx.Kaiko.ApiClient.Tests/Integration/KaikoClientTestsBase.cs

```diff
@@ -1,24 +1,60 @@
-using Microsoft.Extensions.DependencyInjection;
+using System;
+using Serilog;
 
-namespace Trakx.Kaiko.ApiClient.Tests.Integration
+namespace Trakx.Kaiko.ApiClient.Tests.Integration;
+
+[Collection(nameof(ApiTestCollection))]
+public class KaikoClientTestsBase
 {
-    public class KaikoClientTestsBase
+    protected readonly ITestOutputHelper Output;
+    protected readonly ILogger Logger;
+    protected readonly IServiceProvider ServiceProvider;
+
+    public KaikoClientTestsBase(KaikoApiFixture apiFixture, ITestOutputHelper output)
     {
-        protected ServiceProvider? ServiceProvider;
+        Output = output;
+        ServiceProvider = apiFixture.ServiceProvider;
+        Logger = new LoggerConfiguration().WriteTo.TestOutput(output).CreateLogger();
+    }
+}
+
+
+[CollectionDefinition(nameof(ApiTestCollection))]
+public class ApiTestCollection : ICollectionFixture<KaikoApiFixture>
+{
+    // This class has no code, and is never created. Its purpose is simply
+    // to be the place to apply [CollectionDefinition] and all the
+    // ICollectionFixture<> interfaces.
+}
+
+public class KaikoApiFixture : IDisposable
+{
+    public ServiceProvider ServiceProvider { get; }
 
-        public KaikoClientTestsBase()
+    public KaikoApiFixture()
+    {
+
+        var configuration = ConfigurationHelper.GetConfigurationFromAws<KaikoApiConfiguration>()
+            with
         {
-            var configuration = new KaikoApiConfiguration
-            {
-                ApiKey = Secrets.KaikoApiKey
-            };
+            BaseUrl = new Uri("https://eu.market-api.kaiko.io")
```

Comment:
Should this URL move to a config class or a config file?

Reply:
Thanks. It definitely should, like I did for Stream tests. Fixing.